astrobee/survey_manager/survey_planner/readme.md

```diff
@@ -26,4 +26,142 @@ Don't forget to change the commands to use the target robot's name (bumble is wh
+instructions for NASA-internal users both for
+[`astrobee`](https://github.com/nasa/astrobee/blob/develop/doc/general_documentation/NASA_INSTALL.md)
+and [`isaac`](https://github.com/nasa/isaac/blob/develop/INSTALL.md).
+
+Set up GDS and `astrobee_ops` per [this NASA-internal wiki
+page](https://babelfish.arc.nasa.gov/confluence/display/FFFSW/Running+GDS+in+Linux). You
+can stop when you get to the "Setup Sci Cam Streaming" section.
+
+### Environment variables to define in all terminals below
+
+These could be added to `~/.bashrc` if you prefer:
+```bash
+OPS_REPO="${HOME}/astrobee/ops"  # Or wherever you put this
+ISAAC_WS="${HOME}/isaac"  # Or wherever you put this
+source "${ISAAC_WS}/devel/setup.bash"
+```
+
+### Running the scenario
+
+1. Clean up any lingering processes from previous runs:
+   ```bash
+   # === In terminal 1 ===
+   ps auxww | grep -P 'ros|survey|astrobee|gzserver|gazebo|inspection_tool'
+   # If you see processes that look relevant, you can kill them with
+   # 'killall <name>' or 'killall -9 <name>' if needed. Failing to clean
+   # up any lingering processes seems to cause unpredictable faults like
+   # robots not moving when commanded.
+   ```
+
+2. Start the simulator:
+   ```bash
+   # === In terminal 1 ===
+   roslaunch isaac sim.launch honey:=true dds:=true gds:=true robot:=sim_pub
+   # roslaunch will run indefinitely. Press Ctrl-C to shut down the system.
+   ```
+
+3. Configure the Astrobees, using one of these options:
+   1. Using the Astrobee Workbench (a.k.a. "GDS") graphical interface:
+      1. Configure the recording profile. Go to the "Engineering" tab
+         and do the following for each robot:
+         1. Use the "Select Bee..." menu at the top left to select the robot.
+         2. Click "Grab Control".
+         3. In the "Data to Disk" panel, select "SurveyNoDepth.json" in the
+            leftmost drop-down menu. If this option is not available,
+            review the "Setup GDS" section of the wiki page referenced
+            above and restart.
+         4. Click the leftmost "Configure Data" button next to the menu.
+      2. Turn off face-forward motion. Go to the "Teleoperate" tab and do
+         the following for each robot:
+         1. Use the "Select Bee..." menu at the top left to select the robot.
+         2. Click "Grab Control".
+         3. In the "Bee Commanding" tab, in the "Options" column,
+            uncheck the purple checkmark to the left of "Face Forward"
+         4. Click "Apply Options". You should see the green checkmark to
+            the right of "Face Forward" turn into a black "X".
+
+   2. Using command-line tools:
+      ```bash
+      # === In terminal 2 ===
+      cat <<EOF >/tmp/configure_cmds.txt
+      # Configure the recording profile
+      for prefix in "" "/honey"; do
+          TOPIC_PREFIX="\${prefix}" "${OPS_REPO}/dock_scripts/hsc/cmd" -c bagger -config "${OPS_REPO}/gds/ControlStationConfig/DataToDisk-ISAAC/SurveyNoDepth.json"
```

Comment:
This is not working yet pending a PR on the `astrobee_ops` repo that adds `$TOPIC_PREFIX` support.

Reply:
The relevant PR is now merged. (But remember to update `astrobee_ops`!)